www/docs/client/mocking.md

```diff
@@ -0,0 +1,234 @@
+
+Now we just need to create a wrapper that exposes the client. This process is similar to [the React setup](/reactjs/introduction.mdx) however we don't need to utilize the `useState` hook because our tests are not using SSR.
+
+```ts title='trpcMockedProvider.ts'
+// @ts-expect-error tRPC utilizes global fetch
+global.fetch = fetch;
+
+const mockedtRPCClient = mockedtRPC.createClient({
+  links: [
+    loggerLink({ enabled: () => false }),
+    httpBatchLink({ url: "http://localhost:3000/api/trpc" }),
+  ],
+});
+
+const mockedQueryClient = new QueryClient();
+
+export const TRPCMockedProvider = (props: { children: React.ReactNode }) => {
+  return (
+    <mockedtRPC.Provider
+      client={mockedtRPCClient}
+      queryClient={mockedQueryClient}
+    >
+      <QueryClientProvider client={mockedQueryClient}>
+        {props.children}
+      </QueryClientProvider>
+    </mockedtRPC.Provider>
+  );
+};
+```
+
+### 3. (Optional) Export a custom render hook
+
+To use this mocked provider easily with `@testing-library/react`, we can export a custom render hook that wraps our rendered component in our new `TRPCMockedProvider`.
+
+```ts title='trpcMockedProvider.ts'
+export const renderWithProviders = (
+  ui: ReactElement,
+  options?: Omit<RenderOptions, "wrapper">
+) => {
+  return render(ui, {
+    wrapper: (props) => <TRPCMockedProvider {...props} />,
+    ...options,
+  });
+};
+```
+
+:::note
+
+- If you already have a similar to render a component with providers in tests, you may choose to import `TRPCMockedProvider` to that function instead, this is a simple example for demonstration purposes.
+- If you don't want to use a custom hook like the one illustrated, you may also choose to simply use the return value of the function.
+
+:::
+
+## Using mocks
+
+Now we have `TRPCMockedProvider`, we can start writing and using mocks. The following examples will use [Mock Service Worker](https://mswjs.io/) to intercept our API calls and return a response, as well as [`msw-trpc`](https://www.npmjs.com/package/msw-trpc) to utilize the type signature of your API when defining our [request handlers](https://mswjs.io/docs/getting-started/mocks/rest-api).
+
+To mock data using your own router, start by installing both packages into your dev dependendies:
+
+`npm install --save-dev msw msw-trpc`
+
+### 1. Create your MSW handlers
+
+To utilize tRPC's typing superpowers, we can create a `mswTrpc` constant which generates a typed set of handlers for MSW, based on she shape of your `AppRouter`. This enables us to build handlers specific to our individual tests rapdily and with typesafety.
+
+```ts title='trpcMockedProvider.ts'
+import type { AppRouter } from "path/to/your/router";
+import { createTRPCMsw } from "msw-trpc";
+
+// YOUR PROVIDER
+
+export const trpcMsw = createTRPCMsw<AppRouter>();
+```
+
+:::note
+
+If you want to skip this step, it is possible to create standard REST handlers using `msw`s `rest` object and defining them to your network calls.
```

Comment:
There is probably a more clear phrasing than "defining them to your network calls" but I can't figure out what it is right now either

Reply:
Agreed. I've tried making it more explicit, here's my rephrasing:

> If you want to skip this step, it is possible to create standard REST handlers using `msw`s `rest` object by manually defining the Request URL and Method corresponding to the tRPC procedure you want to mock.